### PR TITLE
Add github actions for CI test

### PR DESCRIPTION
**Pull request by WeichenXu123**
Add github actions for CI test

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Main
+on: [push]
```
Comment (harupy) [inline .github/workflows/main.yml:2]: can you trying adding `pull_request` here?

Comment (WeichenXu123) [reply .github/workflows/main.yml:2]: `on [push, pull_request]` ?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+          architecture: x64
+      - run: |
+          pip install joblib==0.14.0 scikit-learn==0.23.1 nose==1.3.3 pytest pylint
+          pip install ${{ matrix.PIP_PYSPARK }}
+          ./run-tests.sh
```
Comment (harupy) [inline .github/workflows/main.yml:20]: We should run tests in a separate step. Doing both installation and tests makes it difficult to debug.

File: test/test_spark.py [hunk 3]
```diff
@@ -30,6 +30,8 @@
 from sklearn.model_selection import cross_val_score
 from sklearn import datasets
 from sklearn import svm
+import pyspark
+import pyspark.util
```
Comment (harupy) [inline test/test_spark.py:34]: Do we use `pyspark.util` in this file?

File: run-pylint.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Return on any failure
+set -e
+
+# The current directory of the script.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (harupy) [inline run-pylint.sh:24]: Do we need this line?

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+          architecture: x64
+        run: |
+          pip install joblib==0.14.0 scikit-learn==0.23.1 nose==1.3.3 pytest pylint
+          pip install ${{ matrix.PIP_PYSPARK }}
```
Comment (harupy) [inline .github/workflows/main.yml:19]: Are you sure this is correct syntax?

File: run-tests.sh [hunk 6]
```diff
@@ -23,15 +23,5 @@ set -e
...
 # Run test suites
-echo python -m "nose" -v --all-modules -w $DIR 2>&1 | grep -vE "INFO (ParquetOutputFormat|SparkContext|ContextCleaner|ShuffleBlockFetcherIterator|MapOutputTrackerMaster|TaskSetManager|Executor|MemoryStore|CacheManager|BlockManager|DAGScheduler|PythonRDD|TaskSchedulerImpl|ZippedPartitionsRDD2)";
-
-python -m "nose" -v --all-modules -w $DIR 2>&1 | grep -vE "INFO (ParquetOutputFormat|SparkContext|ContextCleaner|ShuffleBlockFetcherIterator|MapOutputTrackerMaster|TaskSetManager|Executor|MemoryStore|CacheManager|BlockManager|DAGScheduler|PythonRDD|TaskSchedulerImpl|ZippedPartitionsRDD2)";
-
-# Exit immediately if the tests fail.
-# Since we pipe to remove the output, we need to use some horrible BASH features:
-# http://stackoverflow.com/questions/1221833/bash-pipe-output-and-capture-exit-status
-test ${PIPESTATUS[0]} -eq 0 || exit 1;
+pytest test -s
```
Comment (harupy) [inline run-tests.sh:27]:
```suggestion
pytest test -s --color=yes
```

**Review by harupy — APPROVED** [review]
LGTM once my comments have been addressed

File: test/test_spark.py [hunk 7]
```diff
@@ -92,8 +94,8 @@ def test_fn(x):
                 pass
 
     with pytest.raises(Exception):
-        with parallel_backend('spark', n_jobs=4):
-            Parallel()(delayed(test_fn)(i) for i in range(4))
+        with parallel_backend('spark', n_jobs=2):
+            Parallel()(delayed(test_fn)(i) for i in range(2))
```
Comment (harupy) [inline test/test_spark.py:98]: Not the scope of this PR, we should add `match`to `pytest.raises`:

```python
with pytest.raises(Exception, match="..."):
    ...
```

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+          architecture: x64
+      - name: Install python packages
+        run: |
+          pip install joblib==0.14.0 scikit-learn==0.23.1 nose==1.3.3 pytest pylint
```
Comment (harupy) [inline .github/workflows/main.yml:19]: Do we still need `nose`?

File: test/test_spark.py (hunk 7 above)
Comment (WeichenXu123) [reply test/test_spark.py:98]: The exception message is different between pyspark 3.2 and lower pyspark versions. We can improve it in followup PR

**Pull request merged**